pkg/predicate/toolchainconditionchanged.go

```diff
@@ -0,0 +1,62 @@
+package predicate
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var _ predicate.Predicate = (*ToolchainConditionChanged)(nil)
+
+type (
+	GetToolchainConditions func(client.Object) []toolchainv1alpha1.Condition
+
+	// ToolchainConditionChanged is a predicate to check that some conditions changed in a toolchain object.
+	ToolchainConditionChanged struct {
+		// GetConditions is function used to obtain the conditions from an object. There is no generic way of
+		// doing that using the standard k8s api.
+		GetConditions GetToolchainConditions
+
+		// Type is the type of the condition to look at
+		Type toolchainv1alpha1.ConditionType
+	}
+)
+
+// Create implements predicate.Predicate.
+func (t *ToolchainConditionChanged) Create(event.CreateEvent) bool {
+	return true
+}
+
+// Delete implements predicate.Predicate.
+func (t *ToolchainConditionChanged) Delete(event.DeleteEvent) bool {
+	return true
+}
+
+// Generic implements predicate.Predicate.
+func (t *ToolchainConditionChanged) Generic(event.GenericEvent) bool {
+	return true
+}
+
+func (t *ToolchainConditionChanged) Update(evt event.UpdateEvent) bool {
```

Comment:
The update event could ideally also cover the create and delete event too,(which probably could in turn reduce the reconciles) but that would complicate the logic and probably not worth the effort just to cover the edge case
is my understanding correct here?

Reply:
I agree. The TCs are not created/delete that often though, so I think we're fine with doing a couple more reconciles than absolutely necessary.